sql/rowexec/join_iters.go

```diff
@@ -669,3 +669,202 @@ func (i *crossJoinIterator) Close(ctx *sql.Context) (err error) {
 
 	return err
 }
+
+// lateralJoinIter is an iterator that performs a lateral join.
+// A LateralJoin is a join where the right side is a subquery that can reference the left side, like through a filter.
```

Comment:
I'm not sure a person new to lateral joins would be have all of their confusion resolved by this docstring